File: JavaScript/938-Range Sum of BST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var rangeSumBST = function(root, low, high) {
    res = 0
    
    let dfs = (root) => {
        if (!root) return;
        if (low <= root.val && root.val <= high){
            res += root.val;
        }
        if (root.val > low){
            dfs(root.left);
        }
        if (root.val < high){
            dfs(root.right);
        }
    }
    dfs(root);
    return res;
};

File: C++/938-Range Sum of BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res = 0;
    void dfs(TreeNode* node, int low, int high) {
        if (node == nullptr) return;
        int val = node->val;
        if (low <= val && val <= high) {
            res += val;
            dfs(node->left, low, high);
            dfs(node->right, low, high);
        } else if (val < low){
            dfs(node->right, low, high);
        } else {
            dfs(node->left, low, high);
        }
    }
    
    int rangeSumBST(TreeNode* root, int low, int high) {
        dfs(root, low, high);
        return res;
    }
};